pkg/controller/restore/restore_controller.go

```diff
@@ -155,6 +157,37 @@ func (c *Controller) updateRestore(cur interface{}) {
 		return
 	}
 
+	if v1alpha1.IsRestoreDataComplete(newRestore) {
+		tc, err := c.getTC(newRestore)
+		if err != nil {
+			klog.Errorf("Fail to get tidbcluster for restore %s/%s, %v", ns, name, err)
```

Comment:
why not enqueue the restore if err != nil

Reply:
Restore requires the tidbcluster exists, otherwise the restore is meaningless.

Reply:
yes, i think that may be ok. this rarely happens and if this happens, it can be retried by informer, beacuse operator set `resyncPeriod` when init `SharedIndexInformer` [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/client/informers/externalversions/pingcap/v1alpha1/restore.go#L56)

Reply:
if err != nil, it also cloud be the network problem, we should retry it. As operator has `resyncPeriod` and can retry after `resyncPeriod` automatically, I think not enqueueing is OK.